### PR TITLE
[Improvement] migrates to react-select: selection dropdown component

### LINKED ISSUE #2321: Migrate Selection dropdown component library
### Description
Migrate **UI/Select**  dropdown component to `react-select` from `react-select-search`

**Select** component from `UI/` is being used in a lot of parts and in dynamic form and widget inspector we are using it from Ui/select
Moving all styles and using the same component will reduce duplication of code

### PR DESCRIPTION
**Pull request by arpitnath**
Resolves #2321 

- **Migrates to `react-select` lib from `react-select-search`**
- **Default common styles for the selection dropdown** to avoid duplications
- **Updates selection dropdown component** :
      - widget inspector [widget definations]
      - dynamic form - queries
      - dynamic form - datasource modal
      - datasource: oauth and authentication dropdowns (source component)
      - query pane - query selector dropdown
      - query pane - rest API methods selector

### CODE REVIEW

**Comment by arpitnath** [comment]
stripe query's selection dropdown will be fixed in https://github.com/ToolJet/ToolJet/pull/2106

File: frontend/src/_ui/Select/SelectComponent.jsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+import Select from 'react-select';
+import defaultStyles from './styles';
+
+export const SelectComponent = ({ options = [], value, onChange, ...restProps }) => {
+  const darkMode = localStorage.getItem('darkMode') === 'true';
+  const { styles, hasSearch = false, height, width, placeholder = 'Select..' } = restProps;
+
+  const useStyles = styles || defaultStyles(darkMode, width, height);
+  const selectOptions =
+    options.length === 0
```
Comment (gondar00) [inline frontend/src/_ui/Select/SelectComponent.jsx:12]:
```suggestion
    Array.isArray(options) && options.length === 0
```

File: frontend/src/_ui/Select/SelectComponent.jsx [hunk 2]
```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+import _ from 'lodash';
+import Select from 'react-select';
+import defaultStyles from './styles';
+
+export const SelectComponent = ({ options = [], value, onChange, ...restProps }) => {
+  const darkMode = localStorage.getItem('darkMode') === 'true';
+  const { styles, hasSearch = false, height, width, placeholder = 'Select..' } = restProps;
+
+  const useStyles = styles || defaultStyles(darkMode, width, height);
```
Comment (gondar00) [inline frontend/src/_ui/Select/SelectComponent.jsx:10]:
```suggestion
  const useStyles = !_.isEmpty(styles) ? styles : defaultStyles(darkMode, width, height);
```

cos `{} || ...` would return `{}` & `[] || ...` would return `[]` would miss out on default styles if styles are passed as one of those

File: frontend/src/_ui/Select/SelectComponent.jsx [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+      : options.map((option) => {
+          if (!option.hasOwnProperty('label')) {
+            return _.mapKeys(option, (value, key) => (key === 'value' ? key : 'label'));
+          }
+          return option;
+        });
+
+  const currentValue = typeof value === 'string' ? selectOptions.find((option) => option.value === value) : value;
+
+  console.log('SelectComponent', currentValue);
```
Comment (gondar00) [inline frontend/src/_ui/Select/SelectComponent.jsx:23]: ❌

**Review by gondar00 — APPROVED** [review]
🥇

**Comment by arpitnath** [comment]
Todo:
- Event manager
- Table cell types

**Comment by arpitnath** [comment]
> Todo:
> 
> * Event manager
> * Table cell types

We will resolve this in a separate PR
cc @sherfin94

**Review by sherfin94 — APPROVED** [review]

**Pull request merged**